ui/Makefile

```diff
@@ -49,6 +51,10 @@ $(TYPESCRIPT_TARGET): $(shell find $(TS_ROOT))
 	$(NODE_MODULES)/.bin/tsc -p $(TS_ROOT)
 	$(NODE_MODULES)/.bin/tslint -c $(TS_ROOT)/tslint.json $(addprefix -f , $(shell find $(TS_ROOT) -name *.ts -not -path *typings*))
 
+$(STYLUS_TARGET): css/*.styl
+	npm install
```

Comment:
Made into a real target which depends on the npm shrinkwrap file and generates a local timestamp file.  Also added the timestamp filename to .gitignore.